### PR TITLE
new methods, dto and controllers for `CinemaHall`, `Movie` and `Movie…

### PR DESCRIPTION
**Pull request by Mandybur800**
…Session` was implemented

### CODE REVIEW

File: src/main/java/ua/com/conductor/controllers/MovieSessionController.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    @PutMapping
+    public void updateMovieSession(@RequestBody MovieSessionUpdateDto dto) {
+        MovieSession movieSession = mapper.toUpdateEntity(dto);
+        movieSessionService.update(movieSession);
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteMovieSession(@PathVariable Long id) {
+        movieSessionService.delete(movieSessionService.get(id));
+    }
```
Comment (Vedroid) [inline src/main/java/ua/com/conductor/controllers/MovieSessionController.java]: The assignment says: `Remove movie session - DELETE /movie-sessions`. I think that `@RequestBody` is needed here

File: src/main/java/ua/com/conductor/model/dto/MovieSessionUpdateDto.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package ua.com.conductor.model.dto;
+
+public class MovieSessionUpdateDto {
```
Comment (Vedroid) [inline src/main/java/ua/com/conductor/model/dto/MovieSessionUpdateDto.java]: Are you sure you need this DTO? I do not think that you will receive different DTOs from the request

File: src/main/java/ua/com/conductor/controllers/MovieSessionController.java (hunk 1 above)
Comment (Mandybur800) [reply src/main/java/ua/com/conductor/controllers/MovieSessionController.java]: yep, you are right

**Review by Vedroid — APPROVED** [review]

File: src/main/java/ua/com/conductor/controllers/MovieSessionController.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        movieSessionService.add(movieSession);
+    }
+
+    @PutMapping
+    public void updateMovieSession(@RequestBody MovieSessionRequestDto dto) {
+        movieSessionService.update(mapper.toEntity(dto));
+    }
+
+    @DeleteMapping
+    public void deleteMovieSession(@RequestBody MovieSessionRequestDto dto) {
```
Comment (boroda4436) [inline src/main/java/ua/com/conductor/controllers/MovieSessionController.java]: we should receive an Id of a resource we want to delete
See checklist's link. Try to find an example of `HTTP DELETE http://api.example.com/device-management/managed-devices`

File: src/main/java/ua/com/conductor/model/dto/MovieSessionRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package ua.com.conductor.model.dto;
+
+public class MovieSessionRequestDto {
+    private Long id;
```
Comment (boroda4436) [inline src/main/java/ua/com/conductor/model/dto/MovieSessionRequestDto.java]: How we can use it?

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/ua/com/conductor/model/dto/MovieSessionRequestDto.java (hunk 4 above)
Comment (Mandybur800) [reply src/main/java/ua/com/conductor/model/dto/MovieSessionRequestDto.java]: I removed this

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**